eng/scripts/Test-Code.ps1

```diff
@@ -31,6 +32,71 @@ Remove-Item -Recurse -Force $TestResultsPath -ErrorAction SilentlyContinue
 
 $testProjects = @()
 
+function BuildNativeBinaryAndPrepareTests {
+    param(
+        [string[]]$areas = @()
+    )
+
+    # Native AOT compilation only occurs during 'dotnet publish', not 'dotnet build'
+    $nativeBinaryPath = PublishNativeBinary
+
+    Write-Host "Building test project(s)"
+    Invoke-LoggedCommand `
+        -Command "dotnet build" `
+        -AllowedExitCodes @(0)
+
+    CopyNativeBinaryToTestDirs -nativeBinaryPath $nativeBinaryPath -areas $areas
+}
+
+
+function PublishNativeBinary {
+    $runtimeId = [System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier
+    Write-Host "Publishing AzureMcp as native binary for $runtimeId"
+
+    $cliProjectDir = "$RepoRoot/core/src/AzureMcp.Cli"
+
+    Invoke-LoggedCommand `
+        -Command "dotnet publish '$cliProjectDir/AzureMcp.Cli.csproj' -c Release -r $runtimeId /p:BuildNative=true" `
+        -AllowedExitCodes @(0) | Out-Null
+
+    $exeName = if ($runtimeId.StartsWith('win-')) { "azmcp.exe" } else { "azmcp" }
+    $nativeExePath = "$cliProjectDir/bin/Release/net9.0/$runtimeId/publish/$exeName"
+
+    if (-not (Test-Path $nativeExePath)) {
+        Write-Error "Native binary not found at $nativeExePath"
+        exit 1
+    }
+
+    return $nativeExePath
+}
+
+function CopyNativeBinaryToTestDirs {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$nativeBinaryPath,
+        [string[]]$areas = @()
+    )
+    Write-Host "Copying native AzureMcp to test directories"
+
+    if (!$areas) {
+        $testDirectories = Get-ChildItem -Path $RepoRoot -Recurse -Filter "*.LiveTests" -Directory
+    } else {
+        $testDirectories = @()
+        foreach ($area in $areas) {
+            $areaPath = $area -eq 'core' ? "$RepoRoot/core/tests" : "$RepoRoot/areas/$($area.ToLower())/tests"
```

Comment:
This might run into an issue as it appears VirtualDesktop isn't using lowercase naming

https://github.com/Azure/azure-mcp/tree/main/areas/VirtualDesktop

Reply:
My understanding is that all root directory names for areas are expected to be in lowercase. [Here](https://github.com/Azure/azure-mcp/blob/294c875a9c0209eb370627ea88e82c7141530471/eng/scripts/Test-Code.ps1#L47) is how we prepare for testing today, so I followed this convention for consistency. 

The `Areas` parameter can be set by developers via the command line, as well as by CI if it want to limit test run to certain areas based on code changes.

Great catch that `VirtualDesktop` violates this. Let me do a follow up pr to fix the `VirtualDesktop` naming, so this pr is focused on aot specific.

Reply:
@alzimmermsft, fyi - the owner fixed casing for VirtualDesktop in the main.